apps/asn1parse.c

```diff
@@ -216,6 +216,9 @@ int asn1parse_main(int argc, char **argv)
                 i = BIO_read(in, &(buf->data[num]), BUFSIZ);
                 if (i <= 0)
                     break;
+                /* make sure num doesn't overflow */
+                if ((num + i) <= 0)
+                    goto end;
```

Comment:
Signed integer overflow can potentially be undefined behaviour, even though unsigned integer overflow cannot (per the spec).

Given that, would it be better to rewrite this as `if (i > LONG_MAX - num) ...`?